analyze_issue_metadata.py

```diff
@@ -0,0 +1,186 @@
+
+    Returns:
+        A JSON of issue metadata
+
+    Raises:
+        N/A
+    """
+
+    org = str(sys.argv[1])
+
+    with open(sys.argv[2]) as f:
+        repos = f.readlines()
+    repos = [x.strip() for x in repos]
+
+    # Authenticate with GitHub using Personal Access Token
+    g = github3.login(token=os.environ['GH_TOKEN'])
+
+    data = gim.get_issue_metadata(g, org, repos, state)
+
+    # Dump the data to a JSON file
+    data_json = json.dumps(data)
+    file_time = dt.now().strftime("%Y%m%d")
+    print 'Dumping data to json file'
+    with open('../output_files/' + file_time + '_raw_veneer_issue_meta.json',
+              'w') as f:
+        data = json.dump(data, f, sort_keys=True, indent=4, separators=(
+             ',', ': '))
+
+    return data_json
+
+
+def analyze_issue_metadata(data):
+
+    p0_labels = ['Priority: P0', 'priority: P0', 'priority: p0', 'P0', 'p0']
+    p1_labels = ['Priority: P1', 'priority: P1', 'priority: p1', 'P1', 'p1']
+    p2_labels = ['Priority: P2+', 'priority: P2+', 'priority: p2+',
+                 'P2+', 'p2+', 'Priority: P2', 'priority: P2', 'priority: p2',
+                 'P2', 'p2']
+    fr_question = ['Type: Enhancement', 'Type: Question', 'type: enhancement',
+                   'type: question', 'enhancement']
+
+    analysis = {}
+
+    # Load/decode the data JSON
+    data_decode = json.loads(data)
+
+    for repo in data_decode:
+        if repo == 'updated':
+            continue
+        else:
+
+            analysis[repo] = {'issues': {'count': 0, 'age': 0},
+                              'prs': {'count': 0, 'age': 0},
+                              'p0': {'count': 0, 'age': 0},
+                              'p1': {'count': 0, 'age': 0},
+                              'p2+': {'count': 0, 'age': 0},
+                              'no_priority_label': {'count': 0, 'age': 0},
+                              'fr_question': {'count': 0, 'age': 0}}
+
+            analysis[repo]['issues']['count'] = data_decode[repo][
+                                                  'open_issues_count']
+            analysis[repo]['prs']['count'] = data_decode[repo]['prs'][
+                                               'open_pr_count']
+
+            # if there are prs, calculate their mean age
+            if analysis[repo]['prs']['count'] > 0:
+                analysis[repo]['prs']['age'] = data_decode[repo]['prs'][
+                    'pr_aggregate_age']/analysis[repo]['prs']['count']
+
+            for issue in data_decode[repo]['issues']:
+                created_date = dt.strptime(
+                    data_decode[repo]['issues'][issue]['created'],
+                    "%Y-%m-%dT%H:%M:%S+00:00")
+
+                # it is invalid for an issue to carry multiple priority labels
+                # so count the most severe p0 THEN p1 THEN p2+ THEN fr_question
+
+                if any(x in data_decode[repo]['issues'][issue]['labels']
```

Comment:
I would refactor this into a function rather than a huge switch statement.

```python
def analyze_issue(issue):
    """Return the type of issue that this is deemed to be.

    Args:
        issue (dict): The issue as returned from GitHub. It must have a `labels` key, which
            should be a list.

    Returns:
        str: The type of issue this is.
    """
    # Note: This function relies on the alphabetical ordering of labels, and takes advantage of the
    # fact that "priority" is guaranteed to precede both "type" and "question".
    #
    # If, in the future, it is necessary to evaluate labels with a selection system that does not happen
    # to align with alpha order, this will need to be refactored to do the checks for each label type first,
    # with the inner loop iterating over all the labels to check for presence.
    for label in sorted(issue['labels']):
        # Repositories use different capitalization schemes. Ignore them and do case-insensitive checks.
        label = label.lower()
        if label.endswith('p0'):
            return 'p0'
        if label.endswith('p1'):
            return 'p1'
        if re.search(r'p2\+?$', label):
            return 'p2+'
        if label.endswith('question'):
            return 'fr_question'
        return 'no_priority_label'
```